### PR TITLE
DBZ-6871 MongoDB connector is now able to skip events for oversized documents

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-6871

Should be merged only after https://github.com/debezium/debezium/pull/4834

### CODE REVIEW

**Comment by Naros** [comment]
@jcechace do we need to exclude the new tests from older versions of MongoDB?

**Comment by jcechace** [comment]
@Naros Looking into it...I guess since this feature has to be explicitly enabled it's OK to condition just the tests -- we will simply document that this shoudl be enabled only for MongoDB > 4.4

**Comment by jcechace** [comment]
@jpechane @Naros do we want to still support Mongo 4.4 with 2.4? It's EOL since February and I would like to kepe the policy of supporting just one EOLed version (5.0 goes EOL in October)

**Comment by jpechane** [comment]
@jcechace Jira issue and kill it please

**Comment by jpechane** [comment]
@jcechace Could you please rebase/updated the PR?

**Comment by jpechane** [comment]
@jcechace Applied, thanks a lot! Please prepare a new docs PR with all the new config options. One commit for 2.3/2.4 changes and one for 2.4 only

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java [hunk 1]
```diff
@@ -738,6 +802,27 @@ public static CursorPipelineOrder parse(String value, String defaultValue) {
...
+                    + "'fail' (the default) the connector fails if the total size of change event exceed the maximum BSON size"
+                    + "'skip' any change events for documents exceeding the maximum size will be ignored ");
+
+    public static final Field CURSOR_OVERSIZE_SKIP_THRESHOLD = Field.create("cursor.oversize.skip.threshold")
+            .withDisplayName("Oversize document skip threshold")
+            .withType(Type.INT)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.LOW)
+            .withDefault(0)
+            .withValidation(MongoDbConnectorConfig::validateOversizeSkipThreshold)
```
Comment (sharanrudresh) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java:821]: @jcechace @jpechane These two configs were not added to the list in `CONFIG_DEFINITION` (line 867 below), and thus this validation does not run.
Curious - was this a miss, or was it intentional. I see there's more such configs in the latest `main` branch as well - https://github.com/debezium/debezium/blob/aef8b3eac54bcbe8c07587270e1ec55200eac7b5/debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java#L1019

Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java:821]: @sharanrudresh This is most probably oversight, could you pease add the missing options?

Comment (sharanrudresh) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java:821]: Sure, thankyou!

**Pull request merged**